app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  after_save :update_user_posts_counter
+
+  private
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def recent_comments
+    comments.order(created_at(Desc)).limit(5)
+  end
```

Comment:
Thank you @Reem-lab but  I am slightly confused what I should be doing in regards of this private method. it is not specified in the requirements to make it private, but first code reviewer said it is a required change. now you say it is recommended but not a requirement. moving forward with this project I am scared that once again I will have to implement required changes by next reviewer to make it public as it was not part of the requirements. could you advise me what I should do here?

Reply:
- Hello @fatima-najafi once again 😊, I heavily apologize if our comments have confused you and you're correct it's not mentioned in the project requirements to make the counter methods private, but it's also not mentioned to use Rails Active Callbacks which you seem to be using in your codebase and that's the only reason I suggested you make the methods private. As I explained previously the `update_user_posts_counter` method will be invoked each time a new post is saved to the database due to its callback association this means this method will be used internally and as you learned in OOP principles we need to encapsulate such methods by declaring them as private ensuring you don't accidentally call them and make duplicate increments to the database.
- Regarding the `recent_comments` method keep in mind I only mentioned making the counter methods (like the `update_user_posts_counter`, `update_post_likes_counter` & `update_post_comments_counter`) private as they're only being used internally by the class but the `recent_comments` or any other method not mentioned will be used outside the class so it should be declared as `public`. 


🌠    As code reviewers, our role is to provide guidance for improvement rather than offering a comprehensive step-by-step solution. It is essential for your growth and learning to carefully consider each decision and think critically about the choices you make based on the knowledge you have acquired. This process of independent thinking and problem-solving is where true learning occurs and that's why we don't show you each step of the way. We are here to support you by pointing you in the right direction and helping you develop your skills further. Embrace the opportunity to apply your existing knowledge and engage in thoughtful decision-making as you progress.

Have an awesome day & happy coding! 🎉🎊

Reply:
thank you so much for your explanation @AbelG101